### ISSUE TITLE
[release/2.8] Pytorch dynamo benchmark performance has some dropped compared with 2.7.1

### ISSUE THREAD

**Issue by mengfei25** [intel/torch-xpu-ops#1809]
### 🐛 Describe the bug

Tested with pytorch 2.8 rc3 and release 2.7.1, some models performance has dropped.

Category | Model | inductor [2.8.0/2.7.1]     higher 2.8.0 is better
-- | -- | --
timm_models_float16_training | pit_b_224 | 0.28448831
timm_models_bfloat16_training | pit_b_224 | 0.29032113
timm_models_amp_fp16_training | pit_b_224 | 0.291352859
timm_models_amp_bf16_training | pit_b_224 | 0.295351255
timm_models_float32_training | pit_b_224 | 0.436721557
timm_models_float16_training | crossvit_9_240 | 0.443672806
timm_models_bfloat16_training | crossvit_9_240 | 0.45002951
timm_models_amp_fp16_training | crossvit_9_240 | 0.45858836
timm_models_amp_bf16_training | crossvit_9_240 | 0.467117717
torchbench_float32_inference | hf_T5 | 0.487166425
timm_models_float32_training | crossvit_9_240 | 0.522914784
timm_models_bfloat16_training | swin_base_patch4_window7_224 | 0.581845671
timm_models_float16_training | tnt_s_patch16_224 | 0.589716973
timm_models_bfloat16_training | levit_128 | 0.599264363
torchbench_bfloat16_training | timm_vision_transformer | 0.602623067
timm_models_amp_fp16_training | vit_base_patch16_224 | 0.605192613
timm_models_float16_training | levit_128 | 0.605491707
torchbench_float16_training | timm_vision_transformer | 0.606063019
timm_models_bfloat16_training | tnt_s_patch16_224 | 0.606607499
torchbench_amp_bf16_training | timm_vision_transformer | 0.607674406
timm_models_float16_training | vit_base_patch16_224 | 0.616597764
timm_models_bfloat16_training | vit_base_patch16_224 | 0.619871261
torchbench_amp_fp16_training | timm_vision_transformer | 0.622206259
timm_models_amp_bf16_training | vit_base_patch16_224 | 0.628427813
timm_models_amp_fp16_training | beit_base_patch16_224 | 0.631840027
timm_models_amp_bf16_training | beit_base_patch16_224 | 0.640884129
timm_models_bfloat16_training | beit_base_patch16_224 | 0.672775756
timm_models_float16_training | deit_base_distilled_patch16_224 | 0.677714225
timm_models_amp_bf16_training | deit_base_distilled_patch16_224 | 0.680891069
timm_models_amp_fp16_training | deit_base_distilled_patch16_224 | 0.682489975
timm_models_bfloat16_training | deit_base_distilled_patch16_224 | 0.685373722
timm_models_float32_training | levit_128 | 0.694789847
timm_models_amp_fp16_training | volo_d1_224 | 0.696052388
timm_models_bfloat16_training | volo_d1_224 | 0.699337193
timm_models_amp_bf16_training | volo_d1_224 | 0.702250428
timm_models_float16_training | swin_base_patch4_window7_224 | 0.703094821
timm_models_float16_training | volo_d1_224 | 0.709435268
timm_models_float32_training | tnt_s_patch16_224 | 0.712331823
torchbench_amp_fp16_training | pytorch_stargan | 0.720098993
torchbench_amp_bf16_training | pytorch_unet | 0.723226423
torchbench_bfloat16_training | pytorch_unet | 0.723871848
torchbench_float16_training | pytorch_unet | 0.728192842
torchbench_amp_bf16_training | Background_Matting | 0.742551781
torchbench_bfloat16_training | Background_Matting | 0.742578323
huggingface_amp_bf16_inference | T5ForConditionalGeneration | 0.742609732
torchbench_bfloat16_training | pytorch_stargan | 0.743474241
torchbench_amp_fp16_training | pytorch_unet | 0.743985397
torchbench_amp_fp16_training | Background_Matting | 0.746794993
huggingface_amp_bf16_inference | T5Small | 0.748722197
torchbench_float16_training | Background_Matting | 0.750732848
torchbench_float16_training | pytorch_stargan | 0.751508105
huggingface_amp_fp16_inference | T5Small | 0.753597657
timm_models_float32_training | volo_d1_224 | 0.753682431
huggingface_amp_fp16_inference | T5ForConditionalGeneration | 0.757169712
torchbench_amp_bf16_training | pytorch_stargan | 0.760474302
torchbench_amp_fp16_training | timm_vision_transformer_large | 0.779026217
torchbench_float16_training | timm_vision_transformer_large | 0.783220643
torchbench_float32_training | timm_vision_transformer | 0.794215655
huggingface_float32_inference | T5ForConditionalGeneration | 0.801529029
huggingface_float32_inference | T5Small | 0.803971534
torchbench_bfloat16_training | timm_vision_transformer_large | 0.806120234
torchbench_amp_bf16_training | timm_vision_transformer_large | 0.808850513
timm_models_float32_training | beit_base_patch16_224 | 0.831658542
torchbench_amp_bf16_inference | hf_T5 | 0.833111419
torchbench_bfloat16_training | pytorch_CycleGAN_and_pix2pix | 0.834543498
torchbench_amp_fp16_inference | hf_T5 | 0.838482869
timm_models_float32_training | vit_base_patch16_224 | 0.842950937
timm_models_float32_training | deit_base_distilled_patch16_224 | 0.845278781
timm_models_amp_fp16_inference | crossvit_9_240 | 0.849573016
timm_models_float32_training | swin_base_patch4_window7_224 | 0.86090889
torchbench_float16_training | pytorch_CycleGAN_and_pix2pix | 0.861415328
torchbench_amp_fp16_training | pytorch_CycleGAN_and_pix2pix | 0.862037716
torchbench_amp_bf16_training | pytorch_CycleGAN_and_pix2pix | 0.862377031
timm_models_amp_bf16_inference | crossvit_9_240 | 0.871373489
torchbench_bfloat16_inference | alexnet | 0.876037106
timm_models_float16_inference | lcnet_050 | 0.877303613
timm_models_bfloat16_inference | lcnet_050 | 0.880615298
timm_models_amp_bf16_training | swin_base_patch4_window7_224 | 0.88071401
torchbench_float32_inference | hf_DistilBert | 0.883906148
torchbench_amp_bf16_inference | alexnet | 0.886944618
timm_models_float32_inference | crossvit_9_240 | 0.888090927
torchbench_float16_inference | alexnet | 0.888715135
timm_models_amp_fp16_inference | lcnet_050 | 0.895340214
torchbench_amp_fp16_inference | alexnet | 0.897772859

Reproducer
```
pip install --pre torch==2.8.0 torchvision torchaudio --index-url https://download.pytorch.org/whl/test/xpu
# pip install torch==2.7.1 torchvision torchaudio --index-url https://download.pytorch.org/whl/xpu
pip install pandas psutil scipy

# Torchbench
git clone https://github.com/pytorch/benchmark
cd benchmark
git checkout e03a63be43e33596f7f0a43b0f530353785e4a59
pip install -r requirements.txt
python install.py <model name>
cd ..

# Huggingface
pip install transformers==4.44.2

# TImm
pip install --no-deps git+https://github.com/huggingface/pytorch-image-models@v1.0.14
pip install $(curl -sSL https://raw.githubusercontent.com/huggingface/pytorch-image-models/v1.0.14/requirements.txt | grep -vE torch)

git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout v2.8.0-rc3  # v2.7.1

# apply patches
git clone https://github.com/intel/torch-xpu-ops
python torch-xpu-ops/.github/scripts/apply_torch_pr.py

python benchmarks/dynamo/torchbench.py --performance --float16 -d xpu \
     -n10 --training --only pit_b_224 --cold-start-latency --backend=inductor
```

### Versions

```
Pytorch: v2.8.0-rc3 and v2.7.1
Device: PVC 1100
Driver: Rolling 25.13.33276
IGC: 2.10.11-1133~22.04
L0: 1.21.1.0-1123~22.04
```

**Comment by jianyizh** [comment]
hf_T5 fp32 inference: onednn graph fallback sdpa, but it's much slower than eager sdpa math path. e.g. reusable_softmax_fwd_generic takes ~5ms but softmax in eager takes less than 1ms

**Comment by jianyizh** [comment]
alex net: triton_poi_fused_convolution_0 become slower, it's a transpose op, check if kernel changed by inductor or it's triton regression

**Comment by LuFinch** [comment]
@jianyizh 

**Q** : hf_T5 fp32 inference: onednn graph fallback sdpa, but it's much slower than eager sdpa math path. e.g. reusable_softmax_fwd_generic takes ~5ms but softmax in eager takes less than 1ms. inductor does not fuse t5 sdpa in 2.7.
**A** : The T5'SDPA pattern is add three weeks ago. Hence inductor will fuse sdpa in 2.8 but not in 2.7.   However, OneDNN SDPA don't support kernel for fp32 hence it fallback to ref path. It is a known issue that ref path is slower than PT Math, which will be fix in OneDNN v3.9

**Q** : For amp, some sdpa does not enter micro sdpa kernel.
**A**:  checked that this is because attention_mask in T5 is FP32 while QKV is FP16 in amp mode. OneDNN SDPA has a check that attention_mask should has the same dtype with QKV, hence it runs into ref path and has bad perf. I have reported this issue to OneDNN and they will remove this check in v3.9.

**Comment by chuanqi129** [comment]
All issues are root caused
